src/graphics/shader/shaderbuilder.cpp

```diff
@@ -273,12 +301,101 @@ ShaderBuilder::~ShaderBuilder() {
 Common::UString ShaderBuilder::genVertexShader(uint32 flags, bool isGL3) {
 	Common::UString header;
 	Common::UString body;
+	Common::UString combined;
+
+	initVertexShaderString(header, body, isGL3);
+	addVertexShaderString(header, body, flags, ShaderBuilder::BLEND_ONE, isGL3);
+	finaliseVertexShaderString(combined, header, body, isGL3);
+	return combined;
+}
+
+Common::UString ShaderBuilder::genVertexShaderName(uint32 flags) {
+	return getBaseShaderName(flags ) + "vert";
+}
+
+Common::UString ShaderBuilder::genFragmentShader(uint32 flags, bool isGL3) {
+	Common::UString header;
+	Common::UString body;
+	Common::UString combined;
+
+	initFragmentShaderString(header, body, isGL3);
+	addFragmentShaderString(header, body, flags, ShaderBuilder::BLEND_ONE, isGL3);
+	finaliseFragmentShaderString(combined, header, body, isGL3);
+	return combined;
+}
+
+Common::UString ShaderBuilder::genFragmentShaderName(uint32 flags) {
+	return getBaseShaderName(flags ) + "frag";
+}
+
+void ShaderBuilder::initShaderName(Common::UString &name) {
+	name = "xoreos.";
+}
+
+void ShaderBuilder::addShaderName(Common::UString &name, uint32 passType, uint32 blendType) {
+	switch (passType) {
+	case ShaderBuilder::ENV_CUBE:
+		name += "env_cube.";
+		break;
+	case ShaderBuilder::ENV_SPHERE:
+		name += "env_sphere.";
+		break;
+	case ShaderBuilder::COLOUR:
+		name += "colour.";
+		break;
+	case ShaderBuilder::TEXTURE:
+		name += "texture.";
+		break;
+	case ShaderBuilder::TEXTURE_LIGHTMAP:
+		name += "texture_lightmap.";
+		break;
+	case ShaderBuilder::TEXTURE_BUMPMAP:
+		name += "texture_bumpmap.";
+		break;
+	case ShaderBuilder::TEXTURE_LIGHTMAP_BUMPMAP:
+		name += "texture_lightmap_bumpmap.";
+		break;
+	default: break;
+	}
+
+	switch (blendType) {
+	case ShaderBuilder::BLEND_SRC_ALPHA:
+		name += "blend_src_alpha.";
+		break;
+	case ShaderBuilder::BLEND_DST_ALPHA:
+		name += "blend_dst_alpha.";
+		break;
+	case ShaderBuilder::BLEND_ZERO:
+		name += "blend_zero.";
+		break;
+	case ShaderBuilder::BLEND_ONE:
+		name += "blend_one.";
+		break;
+	default: break;
+	}
+}
+
+void ShaderBuilder::finaliseShaderNameVertex(Common::UString &name) {
+	name += "vert";
+}
 
+void ShaderBuilder::finaliseShaderNameFragment(Common::UString &name) {
+	name += "frag";
+}
+
+void ShaderBuilder::initVertexShaderString(Common::UString &header, Common::UString &body, bool isGL3) {
 	if (isGL3) {
 		header = header_default_3vert;
 		body = body_default_start_3vert;
+	} else {
+		header = header_default_2vert;
+		body = body_default_start_2vert;
+	}
+}
 
-		switch (flags) {
+void ShaderBuilder::addVertexShaderString(Common::UString &header, Common::UString &body, uint32 passType, uint32 blendType, bool isGL3) {
```

Comment:
The parameter `blendType` is unused in this method (and will throw an "unused parameter" warning). Please wrap `blendType` in the `UNUSED()` macro to document that it is unused (and silence that warning).